support/test-api/src/main/java/com/hotels/styx/testapi/StyxServer.java

```diff
@@ -184,6 +184,42 @@ public HealthCheckRegistry healthCheckRegistry() {
     public static final class Builder {
         private final Map<String, com.hotels.styx.api.service.BackendService> routes = new HashMap<>();
         private final List<PluginFactoryConfig> pluginFactories = new ArrayList<>();
+        private int proxyHttpPort;
+        private int adminHttpPort;
+        private int proxyHttpsPort;
+
+        /**
+         * Specifies the HTTP port for proxy server.
```

Comment:
These methods should have javadoc comments explain that the endpoints will be assigned random free ports if not set explicitly.

Reply:
Side note: Speaking of documentation, we do not have actual documentation for using the test-api yet, but we have an issue #177 regarding this.

Reply:
Ok, I will fix the documentation.

Reply:
Setting it to zero will do it, but it's probably a more likely use case for someone to simply not call the method at all. I would phrase the comment like:
`By default, Styx will automatically allocate a free port number. This happens when a port is not set a value on the builder, or it is set a value of 0 (zero).`